### PR TITLE
[CCR] Add `ccr.auto_follow_coordinator.wait_for_timeout` setting

### PR DESCRIPTION
**Pull request by martijnvg**
This setting controls the wait for timeout the autofollow coordinator
should use when setting cluster state requests to a remote cluster.

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java [hunk 1]
```diff
@@ -28,6 +29,12 @@ private CcrSettings() {
     public static final Setting<Boolean> CCR_FOLLOWING_INDEX_SETTING =
             Setting.boolSetting("index.xpack.ccr.following_index", false, Property.IndexScope, Property.InternalIndex);
 
+    /**
+     * Dynamic node setting for specifying the wait_for_timeout that the auto follow coordinator should be using.
+     */
+    public static final Setting<TimeValue> CCR_AUTO_FOLLOW_COORDINATOR_WAIT_FOR_TIMEOUT = Setting.timeSetting(
+        "ccr.auto_follow_coordinator.wait_for_timeout", TimeValue.timeValueSeconds(60), Property.NodeScope, Property.Dynamic);
```
Comment (dnhatn) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java]: Should we come up with another name which indicates that this is a timeout setting for wait_for_metadata_version?

**Review by dnhatn — COMMENTED** [review]
Change looks good. I left a comment on the name of the setting.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java (hunk 1 above)
Comment (martijnvg) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java]: Is `wait_for_metadata_timeout` a better name?

Comment (dnhatn) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java]: Yeah, it's better :). I think we should also drop `coordinator` in the setting name.

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java [hunk 2]
```diff
@@ -36,7 +43,8 @@ private CcrSettings() {
     static List<Setting<?>> getSettings() {
         return Arrays.asList(
                 XPackSettings.CCR_ENABLED_SETTING,
-                CCR_FOLLOWING_INDEX_SETTING);
+                CCR_FOLLOWING_INDEX_SETTING,
+            CCR_AUTO_FOLLOW_WAIT_FOR_METADATA_TIMEOUT);
```
Comment (dnhatn) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrSettings.java]: nit: format.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @martijnvg.

**Pull request merged**